File: laba1/queue.go
package main
import "fmt"
func createEmptyQueue() *Queue {
	return &Queue{head: nil, tail: nil}
}
func addBackQueue(que *Queue, value string) {
	newNode := &Node{data: value, next: nil}
	if que.tail == nil {
		que.head = newNode
		que.tail = newNode
	} else {
		que.tail.next = newNode
		que.tail = newNode
	}
}
func remFrontQueue(que *Queue) {
	if que.head == nil {
		return
	}
	que.head = que.head.next
	if que.head == nil {
		que.tail = nil
	}
}
func printQueue(que *Queue) {
	if que.head == nil {
		fmt.Println("Очередь пуста")
		return
	}
	curr := que.head
	for curr != nil {
		fmt.Print(curr.data)
		if curr.next != nil {
			fmt.Print(" ")
		}
		curr = curr.next
	}
	fmt.Println()
}

File: laba1/queue.cpp
#include <iostream>
#include "structures.h"
#include "queue.h"
using namespace std;


Node* createNodeQueue(string data)
{
    Node* node = new Node;
    node->data = data;
    node->next = nullptr;
    return node;
}

Queue* createEmptyQueue() {
    Queue* que = new Queue;
    que->head = nullptr;
    que->tail = nullptr;
    return que;
}

void remFrontQueue(Queue* que)
{
	Node* temp = que->head;
	que->head = que->head -> next;
	delete temp;
	
	if (que->head == nullptr) que->tail = nullptr;
}

void addBackQueue(Queue* que, string data)
{
    Node* node = createNodeQueue(data);
    if (que->head == nullptr){
    	que->head = node;
    	que->tail = node;
    	return;
    }
    que->tail->next = node;
    que->tail = node;
}

void printQueue(Queue* que) {
    Node* curr = que->head;
    while (curr != nullptr) {
        cout << curr->data << " ";
        curr = curr->next;
    }
    cout << endl;
}

